bindings/go/oci/tar/top_level_artifacts.go

```diff
@@ -0,0 +1,60 @@
+package tar
+
+import (
+	"context"
+	"sync"
+
+	"github.com/opencontainers/go-digest"
+	ociImageSpecV1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
+)
+
+// TopLevelArtifacts returns the top-level artifacts from a list of candidates.
+// An artifact is considered a top-level artifact if it is not referenced by any other artifact.
+// If there is only one artifact in the candidates, it is automatically considered a top-level artifact.
+// It uses content.Successors to find successors of each artifact.
+// The function returns a slice of top-level artifacts.
+func TopLevelArtifacts(ctx context.Context, fetcher content.Fetcher, candidates []ociImageSpecV1.Descriptor) []ociImageSpecV1.Descriptor {
+	// If there's only one artifact, it's automatically a top-level artifact
+	if len(candidates) <= 1 {
+		return candidates
+	}
+
+	// Build a set of all referenced digests
+	var mu sync.Mutex
```

Comment:
Hmm, I don't think you need another mutex if your calling origin is inside a WaitGroup that already does synced access. 🤔

Reply:
Eh, I think you're right. Sync WaitGroup is only about coordination. It doesn't handle syncing. My bad.